stacks-signer/src/client/stacks_client.rs

```diff
@@ -375,6 +375,22 @@ impl StacksClient {
         Ok(blocks_mined / reward_cycle_length)
     }
 
+    /// Get the current reward cycle and whether the prepare phase has started for the next cycle
+    pub fn get_current_reward_cycle_and_prepare_status(&self) -> Result<(u64, bool), ClientError> {
```

Comment:
This makes me wonder if perhaps `PoxConstants` should get moved to `stacks_common` at some point.

Reply:
I think the bigger issue for stacks-signer is that it needs to construct the PoxConstants struct in the first place. That could be part of the deserialization of `/v2/pox_info`, but it isn't today.